src/server/storage/storageAPI.js

```diff
@@ -293,14 +306,52 @@ const setup = (app: Router, gunPublic: StorageAPI, storage: StorageAPI) => {
    */
   app.get(
     '/user/exists',
-    passport.authenticate('jwt', { session: false }),
     wrapAsync(async (req, res, next) => {
       const { user } = req
 
       res.json({ ok: 1, exists: user.createdDate != null, fullName: user.fullName })
     })
   )
 
+  /**
+   * @api {post} /userExists returns user registration method
+   * @apiGroup Storage
+   *
+   * @apiSuccess {Number} ok
+   * @apiSuccess {Boolean} exists
+   * @apiSuccess {String} fullName
+   * @apiSuccess {String} provider
+
+
+   * @ignore
+   */
+  app.post(
+    '/userExists',
+    wrapAsync(async (req, res, next) => {
+      const { identifier, email, mobile } = req.body
+      const existing = await storage.model
+        .find({
+          createdDate: { $exists: true },
+          $or: [{ identifier }, { email: email && sha3(email) }, { mobile: mobile && sha3(mobile) }]
+        })
+        .lean()
+
+      if (existing.length) {
+        //prefer oldest verified account
+        const sorted = sortBy(existing, ['isVerified', 'createdDate'])
```

Comment:
```suggestion
        const [{ torusProvider, fullName }] = existing
```

Reply:
sort still returns a list and not a single item.
i'm also not sure the database sort is in same order, so i'm just keeping the current working and tested setup